### DISCUSSION TITLE
Best Practices for S3 Bucket Permissions? Outline does not work with 'Block Public Access' on

### DISCUSSION THREAD

**Discussion by arifb** [outline/outline#7212] [Self Hosting]
Hi - we want to stand up a privacy focused install. Having fully public access to the s3 bucket is a non-starter for us but I don't think that's necessary in order to run outline?

Ideally, we'd have Block Public Access for the bucket but don't think outline will work in that case, as described pretty well here https://github.com/outline/outline/issues/2769

Can anyone share best practices around what the bucket policy should look like to lock down the outline bucket as much as possible?

Do we leave public access on, and then lock down the specific folders within the bucket to ensure that only avatars and whatever else is hardcoded as needing public access work?

Thank you.

**Comment by tommoor** [comment]
One option is to avoid S3 altogether and use the `local` storage option now that it exists.

Otherwise right now the best option is to leave public access on and trust Outline to correctly manage the permissions on items, which it does already.
